### PR TITLE
Adding parameter to dragEnter and dragExit

### PR DESCRIPTION
**Pull request by moniika**
## The basics

- [x] I branched from develop
- [x] My pull request is against develop
- [x] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)

### Resolves

n/a

### Proposed Changes

Adds `dragElement` parameter to `dragEnter` and `dragExit` on `IDragTarget`.

#### Behavior Before Change
No change.

#### Behavior After Change
No change.

### Reason for Changes

Allows components, such as the backpack, to check what kind of element is being dragged in `dragEnter` and `dragExit`.

### Test Coverage

Tested in playground by updating Trashcan to only open for blocks.
```js
Blockly.Trashcan.prototype.onDragEnter = function(dragElement) {
  if (dragElement instanceof Blockly.BlockSvg) {
    this.setLidOpen(true);
  }
};
```

### Additional Information

`dragEnter` and `dragExit` is not called often (only at enter and exit, not continuously thourgh the drag).

### CODE REVIEW

**Comment by moniika** [comment]
I'd like to discuss with this PR, the idea of potentially merging `onBlockDrop` and `onBubbleDrop`.

Also, I'm still investigating the compile errors that are causing the CI run failure.

File: core/drag_target.js [hunk 1]
```diff
@@ -39,15 +39,19 @@ Blockly.DragTarget.prototype.getClientRect;
 
 /**
  * Handles when a cursor with a block or bubble enters this drag target.
+ * @param {!Blockly.BlockSvg|!Blockly.IBubble} _dragElement The block or bubble
+ *     currently being dragged.
```
[the remark this answers is not in the document]
Comment (moniika) [reply core/drag_target.js:43]: I've tried adding the extra parentheses and it doesn't appear to change the error.

[the remark this answers is not in the document]
Comment (cpcallen) [reply core/drag_target.js:43]: I would recommend trying to reduce it to a minimal reproduction.  This smells like a compiler bug to me; either you will prove that or you will discover another bit of code that is causing the compiler to disbelieve that Blockly.BlockSvg is acceptable here.

**Review by alschmiedt — APPROVED** [review]
- I couldn't figure out what exactly was wrong with the types, but I played around with it a bit and it looks like the problem was with `Blockly.IBubble`. If I put in `Blockly.Bubble` it threw errors, but at least knew that the block was part of the type. 
- Are you planning on adding dragElement to `onDragOver` as well? 
- I don't have a strong opinion for combining `wouldDeleteBlock` and `wouldDeleteBubble`. I think ideally they would all take in a drag element, however I get that you don't want to do that for performance reasons. I am leaning slightly towards combining them since I think that is one less thing we would want to change later on.

Approved once we figure out why types are broken.

File: core/drag_target.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (moniika) [reply core/drag_target.js]: For some reason it works when I change `Blockly.BlockSvg` to `Blockly.Block`.

**Review by rachel-fenichel — APPROVED** [review]

**Pull request merged**